girder_worker/app.py

```diff
@@ -109,35 +126,54 @@ def girder_before_task_publish(sender=None, body=None, exchange=None,
             job_model = ModelImporter.model('job', 'jobs')
 
             user = headers.pop('girder_user', getCurrentUser())
-            token = headers.pop('girder_token', None)
-
             task_args, task_kwargs = body[0], body[1]
 
             job = job_model.createJob(
-                **{'title': headers.get('girder_job_title', Task._girder_job_title),
-                   'type': headers.get('girder_job_type', Task._girder_job_type),
-                   'handler': headers.get('girder_job_handler', Task._girder_job_handler),
-                   'public': headers.get('girder_job_public', Task._girder_job_public),
+                **{'title': headers.pop('girder_job_title', Task._girder_job_title),
+                   'type': headers.pop('girder_job_type', Task._girder_job_type),
+                   'handler': headers.pop('girder_job_handler', Task._girder_job_handler),
+                   'public': headers.pop('girder_job_public', Task._girder_job_public),
                    'user': user,
                    'args': task_args,
                    'kwargs': task_kwargs,
                    'otherFields': dict(celeryTaskId=headers['id'],
-                                       **headers.get('girder_job_other_fields',
+                                       **headers.pop('girder_job_other_fields',
```

Comment:
It's confusing that we initially make a distinction between options and headers (in the declaration), then we conflate the 2 into a single dictionary named headers, and pop all of the options out of headers to only have the real headers remain. I think we should just have 2 dictionaries.